test/marketplace/MarketPlace.sol

```diff
@@ -64,12 +63,13 @@ contract MarketPlace {
     string memory n,
     string memory s,
     uint8 d
-  ) public onlyAdmin(admin) returns (bool) {
-    require(swivel != address(0), 'swivel contract address not set');
+  ) public onlyAddress(admin) returns (bool) {
+    address swivelAddr = swivel;
+    require(swivelAddr != address(0), 'swivel contract address not set');
     // TODO can we live with the factory pattern here both bytecode size wise and CREATE opcode cost wise?
     address zctAddr = address(new ZcToken(u, m, n, s, d));
-    address vAddr = address(new VaultTracker(m, c, swivel));
-    markets[u][m] = Market(c, zctAddr, vAddr);
+    address vAddr = address(new VaultTracker(m, c, swivelAddr));
+    markets[u][m] = Market(c, zctAddr, vAddr, 0);
```

Comment:
there's a 4th arg?

Reply:
Yea, one change involved removing the entire mapping for both `matured` and `maturityRate`, folding `maturityRate` into the market.

This lets us check if the market has matured by comparing `maturityRate` to the base case of 0, and also saves gas from using 1 mapping instead of 3.